tests/test_default_qubit.py

```diff
@@ -653,42 +671,12 @@ def test_sample_values(self, qubit_device_2_wires, tol):
 
         qubit_device_2_wires.apply('RX', wires=[0], par=[1.5708])
 
-        s1 = qubit_device_2_wires.sample('PauliZ', [0], [], 10)
+        s1 = qubit_device_2_wires.sample('PauliZ', [0], [])
 
         # s1 should only contain 1 and -1, which is guaranteed if
         # they square to 1
         assert np.allclose(s1**2, 1, atol=tol, rtol=0)
 
-    def test_sample_exception_analytic_mode(self, qubit_device_2_wires):
```

Comment:
I get removing this and the following test, but is there an equivalent set of tests for the default.qubit device?

Reply:
I don't understand what you mean, what kind of test do you think should be there?

Reply:
As a result of this PR, I think this particular edge case could only be achieved via a device such as 
```python
dev = qml.device("default.qubit", wires=2, analytic=False, shots=0)
```
We should add some logic to the base device class to disallow this combination, perhaps.

Reply:
:+1:

Reply:
I mean shots=0 is forbidden anyways, so this should not come up, no?

Reply:
good point, captured by another test :+1: